admin-dev/themes/new-theme/js/pages/product/components/categories/category-tree-selector.ts

```diff
@@ -0,0 +1,471 @@
+
+export default class CategoryTreeSelector {
+  eventEmitter: EventEmitter;
+
+  selectedCategories: Array<Category>
+
+  treeCategories: Array<TreeCategory>
+
+  typeaheadCategories: Array<TypeaheadCategory>;
+
+  defaultCategoryId: number|null;
+
+  modalContainer: HTMLElement|null
+
+  categoryTree: HTMLElement|null;
+
+  prototypeTemplate: string|null;
+
+  prototypeName: string|null
+
+  expandAllButton: HTMLElement|null;
+
+  reduceAllButton: HTMLElement|null;
+
+  tags: Tags|null;
+
+  constructor(eventEmitter: EventEmitter) {
+    this.eventEmitter = eventEmitter;
+    this.treeCategories = [];
+    this.typeaheadCategories = [];
+    this.selectedCategories = [];
+    this.defaultCategoryId = null;
+    this.modalContainer = null;
+    this.categoryTree = null
+    this.prototypeTemplate = null;
+    this.prototypeName = null;
+    this.expandAllButton = null;
+    this.reduceAllButton = null;
+    this.tags = null;
+  }
+
+  public showModal(selectedCategories: Array<Category>): void {
+    this.selectedCategories = selectedCategories;
+    for (const category of selectedCategories) {
+      if (category.isDefault) {
+        this.defaultCategoryId = category.id;
+        break;
+      }
+    }
+
+    const modalContent = $(ProductCategoryMap.categoriesModalTemplate);
+    // @todo: replace fancybox with Modal after following PR is merged:
+    //    https://github.com/PrestaShop/PrestaShop/pull/25184
+    $.fancybox({
+      type: 'iframe',
+      width: '90%',
+      height: '90%',
+      fitToView: false,
+      autoSize: false,
+      content: modalContent.html(),
+      afterShow: () => {
+        this.initModal();
+      },
+    });
+  }
+
+  private async initModal(): Promise<void> {
+    this.modalContainer = document.querySelector(ProductCategoryMap.categoriesModalContainer) as HTMLElement;
+    this.categoryTree = this.modalContainer.querySelector(ProductCategoryMap.categoryTree) as HTMLElement;
+    this.prototypeTemplate = this.categoryTree.dataset.prototype ? this.categoryTree.dataset.prototype : null;
+    this.prototypeName = this.categoryTree.dataset.prototypeName ? this.categoryTree.dataset.prototypeName : null;
+    this.expandAllButton = this.modalContainer.querySelector(ProductCategoryMap.expandAllButton);
+    this.reduceAllButton = this.modalContainer.querySelector(ProductCategoryMap.reduceAllButton);
+    this.tags = new Tags(
+      this.eventEmitter,
+      `${ProductCategoryMap.categoriesModalContainer} ${ProductCategoryMap.tagsContainer}`,
+    );
+    this.tags.render(this.selectedCategories);
+    this.treeCategories = await getCategories();
+
+    this.initTypeaheadData(this.treeCategories, '');
+    this.initTypeahead();
+    this.initTree();
+    this.listenCancelChanges();
+    this.applyCategoryTreeChanges();
+    this.eventEmitter.on(ProductEventMap.categories.categoryRemoved, (categoryId) => this.unselectCategory(categoryId));
+  }
+
+  private applyCategoryTreeChanges(): void {
```

Comment:
```suggestion
  private listenApplyChanges(): void {
```
It better matches the behaviour of the function, besides `applyCategoryTreeChanges` is also the event name, so it's even more confusing if you use it twice for different purposes